yolov7/detect_ptg.py

```diff
@@ -164,48 +279,37 @@ def detect(opt):
             dset['images'].append(image)
             #img_id = dset.add_image(**image)
 
-            # Predict
-            with torch.no_grad():   # Calculating gradients would cause a GPU memory leak
-                pred = model(img, augment=opt.augment)[0]
-            # Apply NMS
-            pred = non_max_suppression(pred, opt.conf_thres, opt.iou_thres, classes=opt.classes, agnostic=opt.agnostic_nms)
-
-            # Process detections
-            for i, det in enumerate(pred):  # detections per image
-                if not len(det):
-                    continue
-                
-                # Rescale boxes from img_size to img0 size
-                det[:, :4] = scale_coords(img.shape[2:], det[:, :4], img0.shape).round()
-                
-                for *xyxy, conf, cls_id in reversed(det): # center xy, wh
-                    gn = torch.tensor(img0.shape)[[1, 0, 1, 0]]  # normalization gain whwh
-                    norm_xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
-                    cxywh = [norm_xywh[0] * width, norm_xywh[1] * height,
-                            norm_xywh[2] * width, norm_xywh[3] * height] # center xy, wh
-                    xywh = [cxywh[0] - (cxywh[2] / 2), cxywh[1] - (cxywh[3] / 2),
-                            cxywh[2], cxywh[3]]
-
-                    ann_id = ann_id + 1
-                    ann = {
-                        "id": ann_id,
-                        "area": xywh[2] * xywh[3],
-                        "image_id": img_id,
-                        "category_id": cls_id,
-                        "bbox": xywh,
-                        "confidence": float(conf),
-                    }
-                    dset['annotations'].append(ann)
-                    #dset.add_annotation(**ann)
-
-                    # Optionaly draw results
-                    if opt.save_img:  # Add bbox to image
-                        label = f'{names[int(cls_id)]} {conf:.2f}'
-                        plot_one_box(xyxy, img0, label=label, color=colors[int(cls_id)], line_thickness=1)
-            
+            gn = torch.tensor(img0.shape)[[1, 0, 1, 0]]  # normalization gain whwh
+            for xyxy, conf, cls_id in predict_image(
+                img0, device, model, stride, imgsz, half, opt.augment,
+                opt.conf_thres, opt.iou_thres, opt.classes, opt.agnostic_nms
+            ):
+                norm_xywh = (xyxy2xywh(xyxy.view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
+                cxywh = [norm_xywh[0] * width, norm_xywh[1] * height,
+                         norm_xywh[2] * width, norm_xywh[3] * height]  # center xy, wh
+                xywh = [cxywh[0] - (cxywh[2] / 2), cxywh[1] - (cxywh[3] / 2),
+                        cxywh[2], cxywh[3]]
+
```

Comment:
can we replace this with
`xywh = (
                        kwimage.Boxes([xyxy], "tlbr")
                        .toformat("xywh")
                        .data[0]
                        .tolist()
                    )`?

Reply:
also a bit of a nitpick, but can we rename `xyxy` to `tlbr` to make it more clear what format it is?

Reply:
This would add a dependency to this package, but the naming is harmful to do.